### PR TITLE
feature: BECS-13 Register

### PR DESCRIPTION
**Pull request by nhidh**
Requirement: https://quang-tran.atlassian.net/jira/software/projects/BECS/boards/3?selectedIssue=BECS-12

### CODE REVIEW

File: src/apps/account/services/user.service.ts [hunk 1]
```diff
@@ -12,21 +12,25 @@ import { Service } from 'typedi';
...
   username: string;
-  firstName: string;
-  lastName: string;
   email: string;
-  gender?: string;
+  address: string;
+  city: string;
+  state: string;
+  zipCode: string;
+  isRemember: boolean;
```
Comment (quangtran85) [inline src/apps/account/services/user.service.ts]: It should be `isMember`

File: src/apps/account/dtos/customer/register.dto.ts [hunk 2]
```diff
@@ -9,15 +9,22 @@ export class CustomerRegisterDto {
...
+  city: string;
+
+  @IsOptional()
+  state: string;
+
+  @IsOptional()
+  zipCode: string;
+
+  @IsOptional()
+  isRemember: boolean;
```
Comment (quangtran85) [inline src/apps/account/dtos/customer/register.dto.ts]: It must be `isMember`

File: src/apps/account/models/user.ts [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
 
-  @prop({ required: true, limit: 50 })
-  public firstName: string;
+  @prop({ required: true, limit: 100 })
+  public email: string;
 
-  @prop({ required: true, limit: 50 })
-  public lastName: string;
+  @prop({ required: false, limit: 100 })
+  public address: string;
```
Comment (quangtran85) [inline src/apps/account/models/user.ts]: This line should be `public address?: string;` due to `required: false`

File: src/apps/account/models/user.ts [hunk 4]
```diff
@@ -1,27 +1,39 @@
...
 
   @prop({ required: false, limit: 100 })
-  public email: string;
+  public city: string;
```
Comment (quangtran85) [inline src/apps/account/models/user.ts]: `public city?: string;`

**Review by quangtran85 — COMMENTED** [review]
I have some comment, plz help to check @nhidh

**Review by quangtran85 — COMMENTED** [review]
I have some comment, plz help to check @nhidh

**Comment by nhidh** [comment]
@quangtran85 : I fixed for your comments. Please help me review again! Thanks

**Review by quangtran85 — APPROVED** [review]
LGTM!

**Pull request merged**